### PR TITLE
Update all gradle files to latest Android Studio 3.0

### PR DESCRIPTION
**Pull request by ViliusKraujutis**
Opening each project with latest stable Android Studio 3.0 is always suggesting to update:
* Gradle version to 4.1
* Android plugin for Gradle
* SDK versions
* Android (support) versions

This would resolve an issue other students had, like #220.

@dnbit, @ceruleanotter could you please review this?

### CODE REVIEW

**Comment by ViliusKraujutis** [comment]
Also, I think it's a good idea to encourage everyone to update to latest stable library versions. Especially if Android Studio 3.0 is suggesting these updates.

I would not want thousands of Android Developers to become ignorant to Android Studio update suggestions and train a habit to always close those popups.

File: Lesson01-Favorite-Toys/T01.01-Exercise-CreateLayout/app/build.gradle [hunk 1]
```diff
@@ -20,8 +20,8 @@ android {
 
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
-    compile 'com.android.support:appcompat-v7:25.1.0'
+    compile 'com.android.support:appcompat-v7:27.0.0'
 
     // TODO (7) Remove the ConstraintLayout dependency as we aren't using it for these simple projects
-    compile 'com.android.support.constraint:constraint-layout:1.0.0-beta4'
+    compile 'com.android.support.constraint:constraint-layout:1.1.0-beta3'
```
Comment (rmazajev) [inline Lesson01-Favorite-Toys/T01.01-Exercise-CreateLayout/app/build.gradle:26]: Can we use stable version of constraint-layout, not beta?

File: Lesson01-Favorite-Toys/T01.02-Exercise-DisplayToyList/app/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.2'
+    compileSdkVersion 27
+    buildToolsVersion '27.0.0'
```
Comment (rmazajev) [inline Lesson01-Favorite-Toys/T01.02-Exercise-DisplayToyList/app/build.gradle:5]: As I know Android Studio by default suggests to use 26.0.2, its stable version. API 27 is Android Oreo 8.1, it's not released yet. Can we use stable version?

**Comment by dnbit** [comment]
I am afraid to say that this could lead to some issues due to the new Android Gradle Plugin. Users can update it via the pop up if they want. I prefer to keep it optional for now. This changes will probably not work in AS 2.3 and some people might have not update to AS 3 yet.

Latest does not necessary mean better. There is not such a thing as keeping students ignorant on something here.

Also, this particular repo is very special in the way that it works so merges cannot be done in the usual way with a PR.

File: Lesson01-Favorite-Toys/T01.02-Exercise-DisplayToyList/app/build.gradle (hunk 2 above)
Comment (ViliusKraujutis) [reply Lesson01-Favorite-Toys/T01.02-Exercise-DisplayToyList/app/build.gradle:5]: Ok.

**Comment by rmazajev** [comment]
@dnbit that's make sense.

**Comment by ViliusKraujutis** [comment]
Closing this PR since all those files are generated from `develop`* branches with **flatten** tool.

**Pull request closed without merging**